erpnext/hr/doctype/training_program/training_program.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2017, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Training Program', {
+	refresh: function(frm) {
+		if(frm.doc.status == 'Scheduled') {
+			frm.add_custom_button(__("Training Event"), function () {
+				frm.trigger('make_training_event');
+			});
+		}
+	},
+	make_training_event: function() {
+		frappe.model.open_mapped_doc({
```

Comment:
You don't need a server method for this. If you use dashboard, the training program will automatically be linked based on the field.

Reply:
I see. I should just Delete the button as the Dashboard is already sufficient. Thanks!